src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,232 @@
+package frc.robot.subsystems.shooter;
+
+import static frc.robot.subsystems.shooter.ShooterConstants.*;
+
+import edu.wpi.first.math.interpolation.InterpolatingDoubleTreeMap;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.lib.team3061.drivetrain.Drivetrain;
+import frc.lib.team3061.util.RobotOdometry;
+import frc.lib.team6328.util.TunableNumber;
+import frc.robot.Field2d;
+import frc.robot.subsystems.intake.Intake;
+import java.util.function.BooleanSupplier;
+import org.littletonrobotics.junction.Logger;
+
+public class Shooter extends SubsystemBase {
+
+  private ShooterIO io;
+  private Intake intake;
+  private Alliance alliance = Alliance.Red;
+  private InterpolatingDoubleTreeMap angleTreeMap;
+  private InterpolatingDoubleTreeMap veloTreeMap;
+  private final ShooterIOInputsAutoLogged shooterInputs = new ShooterIOInputsAutoLogged();
+
+  private final TunableNumber topWheelVelocity = new TunableNumber("Shooter/Top Wheel Velocity", 0);
+  private final TunableNumber bottomWheelVelocity =
+      new TunableNumber("Shooter/Bottom Wheel Velocity", 0);
+  private final TunableNumber angle = new TunableNumber("Shooter/Angle", 10.0);
+
+  public BooleanSupplier inStorage = () -> false;
+
+  private boolean autoShooter = false;
+  private boolean hasNote = false;
```

Comment:
For a state machine, states are usually enumerated values; these six booleans could be captured in a single state instance variable of that enumerated type. This is best practice as it ensures that the states are unique (currently, isAimed and aiming could both be set to true).